packages/flutter/lib/src/gestures/velocity_tracker.dart

```diff
@@ -251,3 +251,86 @@ class VelocityTracker {
     return Velocity(pixelsPerSecond: estimate.pixelsPerSecond);
   }
 }
+
+/// A [VelocityTracker] subclass that matches iOS scroll view's fling velocity
+/// calculation.
+///
+/// To obtain a velocity, call [getVelocity] or [getVelocityEstimate]. The
+/// esimated velocity is typically used as the initial flinging velocity of a
+/// `Scrollable`, when its drag gesture ends.
+class FreeScrollStartVelocityTracker extends VelocityTracker {
+  /// The velocity estimation uses at most 4 samples. The extra samples are
+  /// there to make the `VelocityEstimate.offset` large enough so the gesture
+  /// can pass the `isFlingGesture` test.
+  static const int _sampleSize = 20;
+
+  final List<_PointAtTime> _touchSamples = List<_PointAtTime>(_sampleSize);
+
+  @override
+  void addPosition(Duration time, Offset position) {
+    assert(() {
+      final _PointAtTime previousPoint = _touchSamples[_index];
+      if (previousPoint == null || previousPoint.time <= time)
+        return true;
+      throw FlutterError(
+        'The position being added ($position) has a smaller timestamp ($time)'
+        'than its predecessor: $previousPoint.'
+      );
+    }());
+    _index = (_index + 1) % _sampleSize;
+    _touchSamples[_index] = _PointAtTime(position, time);
+  }
+
+  // Computes the velocity using 2 adjacent points in history. When index = 0,
+  // it uses the latest point recorded and the point recorded immediately before
+  // it. The smaller index is, the ealier in history the points used are.
+  Offset _previousVelocityAt(int index) {
+    final int endIndex = (_index + index) % _sampleSize;
+    final int startIndex = (_index + index - 1) % _sampleSize;
+    final _PointAtTime end = _touchSamples[endIndex];
+    final _PointAtTime start = _touchSamples[startIndex];
+
+    if (end == null || start == null) {
+      return Offset.zero;
+    }
+
+    final int dt = (end.time - start.time).inMicroseconds;
+    assert(dt >= 0);
+
+    return dt > 0
+      // Convert dt to milliseconds to preserve floating point precision.
+      ? (end.point - start.point) * 1000 / (dt.toDouble() / 1000)
+      : Offset.zero;
+  }
+
+  @override
+  VelocityEstimate getVelocityEstimate() {
+    // The velocity estimate tries to match the initial flinging velocity,
```

Comment:
sounds stronger than what the code seems to be doing. You're just weighing the initial velocity more rather than matching the initial velocity no?

Reply:
I was trying to say this is the velocity of the scroll view not that of the gesture. Reworded.